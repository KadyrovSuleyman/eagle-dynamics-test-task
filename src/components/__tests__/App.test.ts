import { Wrapper, createLocalVue, shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import Vuex, { Store } from 'vuex';
import AppVue from '../App.vue';

jest.mock('../state.ts');

let store: Store<{ isAtLeastOnePlayer: boolean }>;
beforeEach(() => {
  store = new Vuex.Store({
    state: {
      isAtLeastOnePlayer: true,
    },
    mutations: {
      change: () => {
        store.state.isAtLeastOnePlayer = !store.state.isAtLeastOnePlayer;
      },
    },
  });
});

const localVue = createLocalVue();
localVue.use(Vuex);
let wrapper: Wrapper<Vue>;
beforeEach(() => {
  wrapper = shallowMount(AppVue, { localVue, store });
});

// ===============================
it('renders', async () => {
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

it('watchs outer store', async () => {
  expect(store.state.isAtLeastOnePlayer).toBeTruthy();
  expect(wrapper.element.outerHTML).toMatchSnapshot();

  store.commit('change');
  await wrapper.vm.$nextTick();
  expect(store.state.isAtLeastOnePlayer).toBeFalsy();
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});
