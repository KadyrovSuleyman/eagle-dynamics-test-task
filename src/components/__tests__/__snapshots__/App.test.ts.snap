// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div id="app">
  <top-panel-stub></top-panel-stub>
  <players-list-stub></players-list-stub>
  <player-info-stub></player-info-stub>
</div>
`;

exports[`watchs outer store 1`] = `
<div id="app">
  <top-panel-stub></top-panel-stub>
  <players-list-stub></players-list-stub>
  <player-info-stub></player-info-stub>
</div>
`;

exports[`watchs outer store 2`] = `
<div id="app">
  <top-panel-stub></top-panel-stub>
  <!---->
  <!---->
</div>
`;
